output/transform/transform_test.go

```diff
@@ -134,7 +134,7 @@ func TestStatements(t *testing.T) {
 	}
 	expectedJSON, _ := json.Marshal(expected)
 
-	// Query: 1, 0, Plan: 0, 1
+	// Query: 1, 0, Plan: 0, 1 (w/ QueryIdx 0)
```

Comment:
note: the "bug" part is; I forgot that I was using the second query as a plan's query, so the QueryIdx of QueryPlanReference does change (previously, it was using the fixed number `1` here).
Now QueryPlanReference has the right set of combinations.